include/sched/async.h

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (C) 2017, Hao Hou
+ **/
+/**
+ * @brief The async task support
+ * @details The async task is the task that do not actually 
+ *          occupies the worker thread. Instead, it runs with the
+ *          async task thread pool, and when the task is done the 
+ *          async task thread will emit a task event which will be able to 
+ *          wake the pending task up in the scheduler task. <br/>
+ *          This mechamism is useful, if we need to performe some slow operation in
+ *          a servlet. Because this types of task won't block the worker thread
+ **/
+#ifndef __SCHED_ASYNC_H__
+#define __SCHED_ASYNC_H__
+
+/**
+ * @brief The type used to describe a task
+ * @todo Determine if we need a timeout
+ **/
+typedef struct {
+	void*                          data;         /*!< The data used by this task */
+	void*                          thread_data;  /*!< The source task scheduler thread context */
+	void*                          sched_data;   /*!< The additional scheduler data */
+	void*                          servlet_ctx;  /*!< The servlet context data */
+	/**
+	 * @brief Initialize the async task
+	 * @param handle The async task handle
+	 * @param async_buf The async data buffer we need to fill
+	 * @param ctxbuf The servlet context buffer
+	 * @return status code
+	 **/
+	int (*init)(const void* handle, void* async_buf, void* ctxbuf);
+	/**
+	 * @brief Actual execution callback, which we should run from the async processing thread
+	 * @note In this function, we don't have any plumber API access besides async_cntl with the handle
+	 * @param handle The task handle
+	 * @param async_buf The async buffer
+	 * @return status code
+	 **/
+	int (*exec)(const void* handle, void* async_buf);
+	/**
+	 * @brief Do the cleanup after the async task is done
+	 * @param async_buf The async buffer
+	 * @param ctxbuf The context buffer
+	 * @return status code
+	 **/
+	int (*cleanup)(void* async_buf, void* ctxbuf);
```

Comment:
Seems we don't need this, becuase there shouldn't be any async controlling operation allowed at this point. So the scheduler can take this .